lib/improver/utilities/save.py

```diff
@@ -102,6 +103,28 @@ def _order_cell_methods(cube):
     cube.cell_methods = cell_methods
 
 
+def _check_metadata(cube):
+    """
+    Checks cube metadata that needs to be correct to guarantee data integrity
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube to be checked
+
+    Raises:
+        ValueError: if time coordinates do not have the required datatypes
+            and units; needed because values may be wrong
+        ValueError: if numerical datatypes are other than 32-bit (except
+            where specified); needed because values may be wrong
+        ValueError: if cube dataset has unknown units; because this may cause
+            misinterpretation on "load"
+    """
+    check_time_coordinate_metadata(cube)
+    check_datatypes(cube)
+    if cf_units.Unit(cube.units).is_unknown():
+        raise ValueError('{} has unknown units'.format(cube.name()))
+
+
 def save_netcdf(cubelist, filename):
     """Save the input Cube or CubeList as a NetCDF file.
 
```

Comment:
Wouldn't hurt to put something in this docstring to say essential metadata is also checked and errors are raised if any issues are found.

Reply:
Updated docstring.